command/generate.go

```diff
@@ -36,23 +35,28 @@ func (c *generateTask) ParseParams(params map[string]interface{}) error {
 }
 
 func (c *generateTask) Execute(ctx context.Context, comm client.Communicator, logger client.LoggerProducer, conf *model.TaskConfig) error {
-
+	var err error
 	if conf.Task.Execution > 0 {
 		logger.Task().Warning("Refusing to generate tasks on an execution other than the first one")
 		return nil
 	}
+  
 	if err := util.ExpandValues(c, conf.Expansions); err != nil {
 		err = errors.Wrap(err, "error expanding params")
 		logger.Task().Error(err)
 		return err
+  }
+
+	if c.Files, err = util.BuildFileList(conf.WorkDir, c.Files...); err != nil {
+		err = errors.Wrap(err, "problem building wildcard paths")
+		logger.Task().Error(err)
+		return err
 	}
-	if len(c.FilesFilter) != 0 {
-		var err error
-		if c.Files, err = util.BuildFileList(conf.WorkDir, c.FilesFilter...); err != nil {
-			err = errors.Wrap(err, "problem building file lists")
-			logger.Task().Error(err)
-			return err
-		}
+
+	if len(c.Files) == 0 {
```

Comment:
We already check this in `ParseParams`. Can that validation pass and this one fail? You should write an expansion test to check this.

Reply:
ParseParams is guaranteed to be called by Execute, since Render (https://github.com/evergreen-ci/evergreen/blob/60a0dd830fdaef6a41de65e2d0eb85b4c018ae91/agent/command.go#L25) calls ParseParams, and then later the agent calls Execute (https://github.com/evergreen-ci/evergreen/blob/60a0dd830fdaef6a41de65e2d0eb85b4c018ae91/agent/command.go#L91).

Reply:
Okay I looked at this a little more--the reason this is necessary is because BuildFileList will remove files that don't exist, so after BuildFileList is called c.Files can now be length 0. (Specifically TestExecuteFileDoesNotExist breaks, which is probably why attach_artifacts also has this check in Execute.)

Reply:
yeah, this is right.

Reply:
Ah gotcha, thanks.